src/snapred/backend/data/GroceryService.py

```diff
@@ -787,7 +787,12 @@ def fetchGroceryList(self, groceryList: List[GroceryListItem]) -> List[Workspace
                             loader="ReheatLeftovers",
                         )
                 case "diffcal_table":
-                    tableWorkspaceName = self._createDiffcalTableWorkspaceName(item.runNumber, item.version)
+                    # this is wrong, wrong, wrong, this should look up calibration off of a sample run number
+                    if not item.version:
+                        item.version = self.dataService._getVersionFromCalibrationIndex(item.runNumber)
+                    record = self.dataService.readCalibrationRecord(item.runNumber, item.version)
+                    item.runNumber = record.runNumber
+                    tableWorkspaceName = self._createDiffcalTableWorkspaceName(record.runNumber, record.version)
```

Comment:
Could you clarify this comment more?

Reply:
this was giving @mguthriem  errors when he was testing, but not sure if resolved now

Reply:
I haven't tried again since this morning. Is there a new version I can try?